### DISCUSSION TITLE
newbie question, why can't i create a new page?

### DISCUSSION THREAD

**Discussion by raisfeld-ori** [vercel/next.js#63917] [Help] [answered]
### Summary

i ended up using nextjs in a new project of mine, and i've been having a bit of an issue with the creation of new pages.
after making a new project, i made a new file named ```test.tsx``` and tried to make it appear as a route (by following the docs), this is the test code i used:
```
export default function Test() {
    return (
      <div>
        <h1>Test Route</h1>
        <p>This is the test route.</p>
      </div>
    );
  }
```
and here is my file structure:
```
./src
├── app
│   ├── globals.css
│   ├── layout.tsx
│   ├── page.css
│   └── page.tsx
└── pages
    └── test.tsx
```

i expected test to appear on ```localhost:3000/test``` but i just got a 404 message.

### Additional information

```js
i have no idea what further information could help solve this issue, if any extra information is needed, just ask.

(also, i didn't make a minimal reproduction since this is just a new project with /pages/test.tsx``` added to it)
```

**Comment by raisfeld-ori** [comment, marked as answer]
it's supposed to be outside the src directory, not in...

**Comment by MarouaneBenbetka** [comment]
this is an example of my folder structure  for a clean code using , for each route i created a folder with name of the route and a page.tsx inside it you can use the same architecture it should work. 

![image](https://github.com/vercel/next.js/assets/96788629/b873ec29-47f2-4ecb-85ca-e1925fc30887)

if this answer your question mark is as answered .

**Reply by raisfeld-ori** [reply]
i already answered it but forgot to mark it. I didn't know you were supposed to put files outside the src directory
